### ISSUE TITLE
Object being recorded outside of its required zone

### ISSUE THREAD

**Issue by markgoodyear** [blakeblackshear/frigate#3804]
### Describe the problem you are having

I have a set up like this: https://docs.frigate.video/guides/stationary_objects, here's mine: https://d.pr/i/J3UGZL. The `drive` zone is set to track `person` only, and the `drive_entrance` is set to track `car` only.

The issue I'm facing is when I drive through the`drive_entrance` zone into the `drive` zone, often the event will just sit at "In Progess" and I get a continuous stream of update events (~ every 4-5s) for the car. I'm not sure what causes it to end, or why it's still being recorded when it is no longer in the `drive_entrace` zone, but it can last anywhere from 30 min to 5+ hours. I'm not sure if what I'm seeing is the expected behavior, so it'd be great to have some insight on this. If it is the expected behaviour, I'd like to avoid it if possible as it takes up a lot of disk space. I have the motion threshold slightly lower as sometimes I get no new events despite either car or person leaving or arriving, but that didn't really help.

I have very hit and miss notifications (both the blueprint, and a basic custom implementation to alert me on `type: new`), unsure if the above is related to that.

Any help/advice is appreciated.

### Version

0.11.0-c461c9e (RC 2)

### Frigate config file

```yaml
mqtt:
  host: ...
  user: ...
  password: ...

cameras:
  front:
    ffmpeg:
      hwaccel_args: -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p
      inputs:
        - path: ...
          roles:
            - detect
            - rtmp
        - path: ...
          roles:
            - record
    detect:
      width: 704
      height: 480
      fps: 5
    motion:
      threshold: 20 # default 25
      mask: # Covers timestamp, foliage, tree
        - 0,174,0,0,74,0,142,0,149,78,181,76,231,84,138,144,0,229
        - 704,159,704,480,558,480,583,133
        - 705,35,705,0,510,0,510,35
    objects:
      track:
        - person
        - car
      filters:
        person:
          min_area: 800
          min_score: 0.65
        car:
          min_score: 0.65
    zones:
      drive_entrance:
        coordinates: 598,116,601,154,445,124,467,96
        objects:
          - car
      drive:
        coordinates: 573,480,187,480,102,303,86,261,283,118,357,119,445,128,431,150,590,188
        objects:
          - person
    record:
      events:
        required_zones:
          - drive_entrance
          - drive
    snapshots:
      enabled: True
      bounding_box: True
      required_zones:
        - drive_entrance
        - drive
    mqtt:
      required_zones:
        - drive_entrance
        - drive

detectors:
  coral:
    type: edgetpu
    device: usb

record:
  enabled: True
  events:
    retain:
      default: 10
      mode: motion
```

### Relevant log output

```shell
None
```

### Frigate stats

```json
{
  "detection_fps": 8.2,
  "detectors": {
    "coral": {
      "detection_start": 0,
      "inference_speed": 9.38,
      "pid": 216
    }
  },
  "front": {
    "camera_fps": 5.1,
    "capture_pid": 225,
    "detection_fps": 8.2,
    "pid": 222,
    "process_fps": 5.1,
    "skipped_fps": 0
  },
  "service": {
    "latest_version": "0.10.1",
    "storage": {
      "/dev/shm": {
        "free": 133.1,
        "mount_type": "tmpfs",
        "total": 134.2,
        "used": 1.1
      },
      "/media/frigate/clips": {
        "free": 89091.5,
        "mount_type": "ext4",
        "total": 102221.2,
        "used": 8646.2
      },
      "/media/frigate/recordings": {
        "free": 89091.5,
        "mount_type": "ext4",
        "total": 102221.2,
        "used": 8646.2
      },
      "/tmp/cache": {
        "free": 985,
        "mount_type": "tmpfs",
        "total": 1000,
        "used": 15
      }
    },
    "temperatures": {},
    "uptime": 43,
    "version": "0.11.0-c461c9e"
  }
}
```

### Operating system

HassOS

### Install method

Docker Compose

### Coral version

USB

### Any other information that may be helpful

Running in Proxmox on a Beelink GK55 with this setup:

- HAOS VM on and the MQTT server is set up as a HA addon
- Debian LXC with Docker (used the [Docker LXC helper from tteck](https://github.com/tteck/Proxmox#-docker-lxc-)). It uses ~5-10% CPU and ~8% memory. The CPU usage increased from 1-3% when enabling hardware acceleration, if that's helpful to know.

**Comment by NickM-27** [comment]
The behavior you are describing is the expected behavior. With `required_zones` set like you have, zones are required for an event to be saved but once an object enters a zone the event is valid until the object is no longer detected.

If you don't want that all to be recorded then I recommend you reading [about the different recording retention modes](https://docs.frigate.video/configuration/record#what-do-the-different-retain-modes-mean). Right now you have `motion` set for the events retention which mean all 10 minute segments that contain motion while the event is ongoing will be saved as part of the event.

As far as "update events" goes you'll need to elaborate as I am not sure what you are referring to.

**Comment by markgoodyear** [comment]
@NickM-27 Appreciate the fast response! I must have initially misunderstood the Avoiding stationary objects docs as I thought that it'd avoid what I'm seeing. From what you said, would I be correct in thinking the set up is just to stop a new event from happening if it gets lost and detected again? That would make sense now.

I'm not overly fussed on cars being recorded so I'll look at the retention modes. All I need is to be notified there's a car and a ideally an image to go with. Just need to figure my inconsistant notifications next.

Regarding the "update events", I just mean the MQTT events containing `"type": "update"`, which it would whilst keeping track of the object.

**Comment by NickM-27** [comment]
Okay then yes the update events is also expected behavior. The docs are pointing towards that you should use the https://docs.frigate.video/integrations/mqtt#frigateevents for notifications to only be notified when `before.entered_zones` is empty and `after.current_zones` has your car zone. This will only notify the first instant the car entered the zone and not again until the car is missed and then detected again.

**Comment by markgoodyear** [comment]
Thanks. Currently for testing purposes I've a notification for `type.new` and `after.entered_zones > 0` which doesn't always work. Often a car coming straight in wouldn't get a notification despite having an event recorded. Makes me think the car may have gone too quickly to be in `after.entered_zones`, but I've not checked the MQTT events that to know if that was the case. Maybe just need to bump the detect fps.

**Comment by NickM-27** [comment]
> Thanks. Currently for testing purposes I've a notification for `type.new` and `after.entered_zones > 0` which doesn't always work. Often a car coming straight in wouldn't get a notification despite having an event recorded. Makes me think the car may have gone too quickly to be in `after.entered_zones`, but I've not checked the MQTT events that to know if that was the case. Maybe just need to bump the detect fps.

I would say it is likely something else. If Frigate created the event (IE the car was detected in the zone) then frigate will send a message with that change having occurred. But yeah using MQTT explorer to actually see the history and look through each message would make it quite a bit easier.

**Comment by markgoodyear** [comment]
For my custom notifications I'm using node-red, and I've got it to save any MQTT events by id (`<id>.txt`) so I've a log by each event to make debugging easier. I've got this running along side the main blueprint as they've both been equally inconsistant, often one would trigger when the other didn't.

However, since adding the logs I think I've spotted one mistake for my own implementation. Filtering by `type.new` is wrong, just had a car cross the `drive_entrance` zone and `type.new` didn't have anything in the `entered_zones` or `current_zones`, theforefore doesn't get passed the `after.entered_zones > 0` condition to send a notification out. This makes sense as Frigate starts tracking way before the zone. Just need to leave it running for a few days and see what happens now.

A little off the original topic so feel free to close this if needs be, I'll open up a new issue if I still have trouble with notifications.
